src/nvim/vim.h

```diff
@@ -136,6 +136,17 @@ enum { FOLD_TEXT_LEN = 51, };  //!< buffer size for get_foldtext()
 #endif
 
 #define STRCPY(d, s)        strcpy((char *)(d), (char *)(s))  // NOLINT(runtime/printf)
+
+#define STRCPY_LEN(d, s, n) (*((char *)memcpy((d), (s), (n)) + (n)) = NUL, (d))
+#define STPCPY_LEN(d, s, n) (STRCPY_LEN(d, s, n) + (n))
+
+// CMakeLists.txt seems to define _GNU_SOURCE.
+#ifdef _GNU_SOURCE
+# define STPCPY(d, s) stpcpy(d, s)
+#else
+# define STPCPY(d, s) xstpcpy(d, s)
```

Comment:
Do we actually need `xstpcpy` anymore? A quick search suggests that MSVC doesn't have `stpcpy`, so yes.